src/components/stake-modal.js

```diff
@@ -341,17 +358,49 @@ const StakeModal = Form.create()(({ ID, form, onCancel }) => {
                   ]
                 })(
                   <StyledInputNumber
-                    parser={useCallback(s => {
-                      s = s.replace(/(?!^-|\.)\D|\.(?![^.]*$)/g, '')
-                      const index = s.indexOf('.')
+                    parser={useCallback(valueAsString => {
+                      valueAsString = valueAsString.replace(
+                        /(?!^-|\.)\D|\.(?![^.]*$)/g,
+                        ''
+                      )
+                      const index = valueAsString.indexOf('.')
                       return index === -1
-                        ? s
-                        : `${s.slice(0, index)}${s.slice(index, index + 19)}`
+                        ? valueAsString
+                        : `${valueAsString.slice(
+                            0,
+                            index
+                          )}${valueAsString.slice(index, index + 19)}`
                     }, [])}
                     precision={0}
                   />
                 )}
               </Form.Item>
+              {shouldShowMaxStakeWarning ? (
+                <Alert
+                  closable
+                  type="info"
+                  message="Tip"
+                  description={
+                    <>
+                      <p>
+                        In case you stake all PNK you have available, then
+                        voting incoherently, your PNK balance may become lower
+                        than the stake. This removes you from the courts
+                        automatically.
```

Comment:
Instead of 'removes' I would use 'unstakes' to keep jargon consistency.

Reply:
Makes sense. Will update.